src/x509.cpp

```diff
@@ -211,7 +211,10 @@ void X509Certificate::verify(const std::vector<X509Certificate> &trustStore,
 {
     VerificationContext ctx;
     ctx.addTrustedCertificates(trustStore)
-       .addIntermediateCertificates(intermediateCAs);
+       .addIntermediateCertificates(intermediateCAs)
+       .setVerificationCheckTime(getNotBeforeAsn1());
+    verify(ctx);
+    ctx.setVerificationCheckTime(getNotAfterAsn1() - Asn1Time::Seconds(1));
```

Comment:
still need to clarify this -1 second. according to the name and to the brief glance at the openssl code this shouldn't be needed